.github/workflows/CI-pipeline.yml

```diff
@@ -0,0 +1,49 @@
+name: CI-pipeline
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on pull request events for specified branches
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains one job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Prepare an environment
+      - name: Environment setup
+        run: |
+          make venv
+
+      # Compare the test and the result files
+      - name: Comparison
+        run: |
+          source .venv/bin/activate
+          python3 parse-hamnosys.py -sf hamnosys_example.txt -df hamnosys_parsed.txt
```

Comment:
These files names has been selected exclusively for debug purposes, I'll change them after Marta's PR merge

Reply:
PR is merged